### PR TITLE
refactor(test/e2e): AddFolder after Commit

### PR DESCRIPTION
**Pull request by rootulp**
Addresses the suggestion in https://github.com/celestiaorg/celestia-app/issues/3349

## Testing

I'm actually pretty hesitant about this change because IDK how to verify it didn't break tests. I tried invoking `make test-e2e` three time from `main` and it always failed. CI doesn't run test-e2e on this PR so we don't have any signal on if this refactor is safe.

### CODE REVIEW

File: test/e2e/testnets/testnet.go [hunk 1]
```diff
@@ -130,21 +130,22 @@ func (t *Testnet) CreateTxClient(name,
...
-			Str("directory", txsimKeyringDir).
 			Str("name", name).
-			Msg("error adding keyring dir to txsim")
+			Msg("error committing txsim")
 		return err
 	}
 
-	err = txsim.Instance.Commit()
+	// copy over the keyring directory to the txsim instance
+	err = txsim.Instance.AddFolder(txsimKeyringDir, txsimRootDir, "10001:10001")
```
Comment (rach-id) [inline test/e2e/testnets/testnet.go:143]: Maybe just ssh into the txsim instance and check:

- The folder has been mounted correctly
- You can send some transactions using that keyring
- The permissions are correct (not sure the containers which users they use)

If all of this is fine, then I suppose they won't break anything

**Review by evan-forbes — APPROVED** [review]
I'd defer to others with more experience w/ knuu, but LGTM

**Comment by rootulp** [comment]
Everytime I try to test this PR, I hit:

```
$ make test-e2e
--> Running end to end tests
go run ./test/e2e
test-e2e2024/04/22 10:19:43 No particular test specified. Running all tests.
test-e2e2024/04/22 10:19:43 go run test/e2e/*.go <test_name> to run a specific test
test-e2e2024/04/22 10:19:43 Valid tests are: E2ESimple

             2024/04/22 10:19:43 === RUN E2ESimple
             2024/04/22 10:19:43 Running simple e2e test version 4da82ef
INFO[2024-04-22T10:19:43-04:00]knuu/knuu.go:168 LOG_LEVEL: info
WARN[2024-04-22T10:19:43-04:00]knuu/knuu.go:63 KNUU_NAMESPACE is deprecated. Scope overridden to: test
INFO[2024-04-22T10:19:43-04:00]knuu/knuu.go:66 Initializing knuu with scope: test
             2024/04/22 10:19:45 Creating testnets validators
             2024/04/22 10:19:45 Creating account
{"level":"info","name":"alice","pk":"PubKeySecp256k1{031247D47FA836F26FC943CFDBBDFD8541323F7A8F780654666ADD992C3C7191AF}","time":"2024-04-22T10:19:45-04:00","message":"txsim account created and added to genesis"}
             2024/04/22 10:19:45 Setting up testnets
{"level":"info","name":"val0","directory":"/var/folders/y0/dd92_x8x4tlf397xstgwfz_c0000gn/T/val0","time":"2024-04-22T10:19:45-04:00","message":"Creating validator's config and data directories"}
{"level":"info","name":"val1","directory":"/var/folders/y0/dd92_x8x4tlf397xstgwfz_c0000gn/T/val1","time":"2024-04-22T10:19:46-04:00","message":"Creating validator's config and data directories"}
{"level":"info","name":"val2","directory":"/var/folders/y0/dd92_x8x4tlf397xstgwfz_c0000gn/T/val2","time":"2024-04-22T10:19:46-04:00","message":"Creating validator's config and data directories"}
{"level":"info","name":"val3","directory":"/var/folders/y0/dd92_x8x4tlf397xstgwfz_c0000gn/T/val3","time":"2024-04-22T10:19:46-04:00","message":"Creating validator's config and data directories"}
             2024/04/22 10:19:46 Starting testnets
2024/04/22 10:20:47 failed to start testnets: node val0 failed to start: error waiting for instance 'val0-9bc4be5e' to be running: timeout while waiting for instance 'val0-9bc4be5e' to be running
exit status 1
make: *** [test-e2e] Error 1
```

so I'm going to look at Rachid's known issues PR to try to resolve.

**Comment by rootulp** [comment]
I tried the steps to increase resources and test still failed 😢

File: test/e2e/testnets/testnet.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ type Testnet struct {
 
 func New(name string, seed int64, grafana *GrafanaInfo) (*Testnet, error) {
 	identifier := fmt.Sprintf("%s_%s", name, time.Now().Format("20060102_150405"))
-	if err := knuu.InitializeWithIdentifier(identifier); err != nil {
```
Comment (staheri14) [inline test/e2e/testnets/testnet.go:31]: This PR could have been split into two PRs, one for replacing namespace with scope and another for the re-ordering of the calls to the Commit and AddFolder.

**Review by staheri14 — COMMENTED** [review]
I am inspecting the issue, for now, here is the log of validators complaining that:
```
2024-04-23 13:40:40.037	
cp: can't stat '/home/celestia/.celestia-app/*': No such file or directory
2024-04-23 13:39:47.056	
cp: can't stat '/home/celestia/.celestia-app/*': No such file or directory
2024-04-23 13:39:21.032	
cp: can't stat '/home/celestia/.celestia-app/*': No such file or directory
2024-04-23 13:39:07.426	
cp: can't stat '/home/celestia/.celestia-app/*': No such file or directory
```
I'd suggest breaking this PR into two, one for replacing namespace with scope and another for the re-ordering of the calls to the Commit and AddFolder to make it easier to debug and find the root cause. While I suspect the issue is due to the reordering of the AddFolder and Commit, but could also be something related to the namespace to the scope change.

File: test/e2e/testnets/testnet.go (hunk 2 above)
Comment (rootulp) [reply test/e2e/testnets/testnet.go:31]: It was already split into two. I handled the deprecated usage in https://github.com/celestiaorg/celestia-app/pull/3347 which has already been merged. I'll rebase this on `main` to clarify it's not actually new in this PR.

Comment (rootulp) [reply test/e2e/testnets/testnet.go]: Done.

**Comment by rootulp** [comment]
Based on @staheri14 's last message, I kinda think this PR may break the test so I'm not going to continue trying to test this

**Comment by staheri14** [comment]
@jrmanes Your inputs on this would be very helpful, I also run into the same issue when swapping the order of commit and addFolder? What might be potentially missing here? Thanks!

**Comment by cmwaters** [comment]
@rootulp are we able to close this now since I believe this was resolved by https://github.com/celestiaorg/celestia-app/pull/3505

**Comment by staheri14** [comment]
Yes, we can close this as the order of calls to commit and add folder is swapped in #3505, which was the intention of this PR. i.e.,
https://github.com/celestiaorg/celestia-app/blob/6f25001dd625e615232dbc79972404f929f7aa04/test/e2e/testnet/testnet.go#L145-L161
and
https://github.com/celestiaorg/celestia-app/blob/6f25001dd625e615232dbc79972404f929f7aa04/test/e2e/testnet/node.go#L224-L232

**Comment by staheri14** [comment]
@rootulp I've gone ahead and closed the PR since I was the assignee, but please let me know if you have any concerns or if you think it should remain open.

**Pull request closed without merging**